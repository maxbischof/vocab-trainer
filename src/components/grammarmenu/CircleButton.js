import React from 'react'
import styled from 'styled-components'

export default function CircleButton({
  text,
  activeProperty,
  color,
  diameter,
  onClick,
}) {
  const borderSize = (parseInt(diameter) * 1.5 - parseInt(diameter)) / 2
  const fontSize = parseInt(diameter) / 2

  return (
    <Circle
      active={activeProperty === text ? true : false}
      color={color}
      diameter={diameter}
      borderSize={borderSize}
      fontSize={fontSize}
      onClick={() => onClick(text)}
    >
      {text}
    </Circle>
  )
}

const Circle = styled.div`
  height: ${(props) => props.diameter};
  width: ${(props) => props.diameter};
  margin: 5px 0;
  background-color: rgba(${(props) => props.color}, 1);
  border-radius: 50%;
  display: inline-block;
  display: grid;
  align-content: center;
  justify-content: center;
  font-weight: 700;
  font-size: ${(props) => props.fontSize + 'px'};
  background-clip: padding-box;
  border: ${(props) =>
    props.active
      ? props.borderSize + 'px solid rgba(' + props.color + ', 0.44);'
      : '0px solid white'};
  cursor: pointer;
  transition: border 0.5s;
`
