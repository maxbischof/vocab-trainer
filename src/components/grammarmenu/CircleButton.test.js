import React from 'react'
import renderer from 'react-test-renderer'
import CircleButton from './CircleButton'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <CircleButton text="A" active={true} color="rgba(43, 173, 134, 1)" />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
