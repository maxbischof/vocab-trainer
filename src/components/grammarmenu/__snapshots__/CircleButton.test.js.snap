// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdfBwQ fFnOqh"
  color="rgba(43, 173, 134, 1)"
  fontSize={NaN}
  onClick={[Function]}
>
  A
</div>
`;
